src/unix/process.c

```diff
@@ -342,6 +346,274 @@ static void uv__process_child_init(const uv_process_options_t* options,
+    * by the file actions
+    * 2) POSIX_SPAWN_SETSIGDEF: Signals mentioned in spawn-sigdefault in
+    * the spawn attributes will be reset to behave as their default
+    * 3) POSIX_SPAWN_SETSIGMASK: Signal mask will be set to the value of
+    * spawn-sigmask in attributes
+    * 4) POSIX_SPAWN_SETSID: Make the process a new session leader if a
+    * detached session was requested. */
+  flags = POSIX_SPAWN_CLOEXEC_DEFAULT |
+    POSIX_SPAWN_SETSIGDEF |
+    POSIX_SPAWN_SETSIGMASK;
+  if (options->flags & UV_PROCESS_DETACHED) 
+    flags |= POSIX_SPAWN_SETSID;
+  err = posix_spawnattr_setflags(attrs, flags);
+  if (err != 0)
+    goto error;
+
+  /*  Reset all signal the child to their default behavior */
+  sigfillset(&signal_set);
+  err = posix_spawnattr_setsigdefault(attrs, &signal_set);
+  if (err != 0) 
+    goto error;
+
+  /*  Reset the signal mask for all signals */
+  sigemptyset(&signal_set);
+  err = posix_spawnattr_setsigmask(attrs, &signal_set);
+  if (err != 0)
+    goto error;
+
+  return err;
+
+error:
+  (void) posix_spawnattr_destroy(attrs);
+  return err;
+}
+
+int uv__spawn_set_posix_spawn_file_actions(posix_spawn_file_actions_t* actions,
+                                           const uv__posix_spawn_fncs_t* posix_spawn_fncs,
+                                           const uv_process_options_t* options,
+                                           int stdio_count,
+                                           int (*pipes)[2])
+{
+  int fd;
+  int err;
+
+  err = posix_spawn_file_actions_init(actions);
+  if (err != 0) {
+    /* If initialization fails, no need to de-init, just return */
+    return err;
+  }
+
+  /* Set the current working directory if requested */
+  if (options->cwd != NULL) {
+    err = posix_spawn_fncs->file_actions.addchdir_np(actions, options->cwd);
+    if (err != 0)
+      goto error;
+  }
+
+  /* First, dupe any required fd into orbit, out of the range of 
+   * the descriptors that should be mapped in. */
+  for(fd = 0 ; fd < stdio_count; ++fd) {
+    if (pipes[fd][1] < 0)
+      continue;
+    
+    err = posix_spawn_file_actions_adddup2(actions, pipes[fd][1], stdio_count + fd);
+    if (err != 0)
+      goto error;
+  }
+
+  /*  Second, move the descriptors into their respective places */
+  for(fd = 0 ; fd < stdio_count; ++fd) {
+    if (pipes[fd][1] < 0)
+      continue;
+
+    err = posix_spawn_file_actions_adddup2(actions, stdio_count + fd, fd);
+    if (err != 0)
+      goto error;
+  }
+
+  /*  Finally, close all the superfluous descriptors */
+  for(fd = 0; fd < stdio_count; ++fd) {
+    if (pipes[fd][1] < 0)
+      continue;
+    
+    err = posix_spawn_file_actions_addclose(actions, stdio_count + fd);
+    if (err != 0)
+      goto error;
+  }
+
+  /*  Finally process the standard streams as per de documentation */
+  for(fd = 0 ; fd < 3 ; ++fd) {
+    /*  If ignored, open as /dev/null */
+    const int oflags = fd == 0 ? O_RDONLY : O_RDWR;
+    const int mode = 0;
```

Comment:
separate code and variables:
```suggestion
    int oflags;
    const int mode = 0;
    oflags = fd == 0 ? O_RDONLY : O_RDWR;
```